src/internal/components/drag-handle-wrapper/__tests__/drag-handle-wrapper.test.tsx

```diff
@@ -127,43 +135,125 @@ test('hides tooltip on Escape', () => {
+  renderDragHandle({
     directions: { 'block-start': 'active' },
-    tooltipText: 'Click me!',
   });
 
   expect(getDirectionButton('block-start')).not.toBeInTheDocument();
   expect(getDirectionButton('block-end')).not.toBeInTheDocument();
 });
 
-test('shows direction buttons when focus enters the button as result of a key input', () => {
-  const { dragHandle, getDirectionButton } = renderDragHandle({
-    directions: { 'block-start': 'active', 'block-end': 'active' },
-    tooltipText: 'Click me!',
+describe('interactionMode = focus (default)', () => {
+  test('shows direction buttons when focus enters the button', () => {
+    const { dragHandle } = renderDragHandle({
+      directions: { 'block-start': 'active', 'block-end': 'active' },
+    });
+
+    document.body.dataset.awsuiFocusVisible = 'true';
+    dragHandle.focus();
+    expect(getDirectionButton('block-start')).toBeVisible();
+    expect(getDirectionButton('block-end')).toBeVisible();
+    expect(getDirectionButton('inline-start')).not.toBeInTheDocument();
   });
 
-  document.body.dataset.awsuiFocusVisible = 'true';
-  dragHandle.focus();
-  expect(getDirectionButton('block-start')).toBeVisible();
-  expect(getDirectionButton('block-end')).toBeVisible();
-  expect(getDirectionButton('inline-start')).not.toBeInTheDocument();
+  test('hides direction buttons when focus leaves the button', () => {
+    const { dragHandle } = renderDragHandle({
+      directions: { 'block-start': 'active', 'block-end': 'active' },
+      tooltipText: 'Click me!',
+    });
+
+    document.body.dataset.awsuiFocusVisible = 'true';
+    dragHandle.focus();
+
+    expect(getDirectionButton('block-start')).toBeInTheDocument();
+    expect(getDirectionButton('block-end')).toBeInTheDocument();
+
+    fireEvent.blur(dragHandle);
+    expectDirectionButtonHidden('block-start');
+    expectDirectionButtonHidden('block-end');
+  });
 });
 
-test('hides direction buttons when focus leaves the button', () => {
-  const { dragHandle, getDirectionButton } = renderDragHandle({
-    directions: { 'block-start': 'active', 'block-end': 'active' },
-    tooltipText: 'Click me!',
+describe('interactionMode = controlled', () => {
+  test('does not show direction buttons when focus enters the button', () => {
+    const { dragHandle } = renderDragHandle({
+      directions: { 'block-start': 'active', 'block-end': 'active' },
+      interactionMode: 'controlled',
+    });
+
+    document.body.dataset.awsuiFocusVisible = 'true';
+    dragHandle.focus();
+    expect(getDirectionButton('block-start')).toBeNull();
+    expect(getDirectionButton('block-end')).toBeNull();
+    expect(getDirectionButton('inline-start')).toBeNull();
+    expect(getDirectionButton('inline-end')).toBeNull();
   });
 
-  document.body.dataset.awsuiFocusVisible = 'true';
-  fireEvent.focusIn(dragHandle, { relatedElement: document.body });
-  fireEvent.focusOut(dragHandle, { relatedElement: document.body });
```

Comment:
Updated this test because `focusIn` and `focusOut` did not call the handlers. Replaced it with `dragHandle.focus();` and `fireEvent.blur(dragHandle);` + added expectations about the arrow visible state on focus and on blur.